### ISSUE TITLE
"impl Snowflake for i64" timestamp << 23 ?

### ISSUE THREAD

**Issue by gouhengheng** [MrGunflame/snowflaked-rs#3]
# i64 $\approx$ 2*u64
```rust
    fn test_snowflaked() {
        use snowflaked::Generator;

        let mut generator: Generator = Generator::builder()
            .epoch(SystemTime::now())
            .instance(123)
            .build();
        sleep(Duration::from_secs(2));
        println!("u64 id: {}", generator.generate::<u64>()); //print 8439443456
        println!("i64 id: {}", generator.generate::<i64>()); // print 16878383105
    }
```
# Why not 22?

https://github.com/MrGunflame/snowflaked-rs/blob/a9c26bde61e8b577444cbb3b83f9268abfdcec4f/src/lib.rs#L159

```rust
fn from_parts(timestamp: u64, instance: u64, sequence: u64) -> Self {
        let timestamp = (timestamp as i64) << 22;
        let instance = (instance << 12) & BITMASK_INSTANCE;
        timestamp + instance as i64 + sequence as i64
    }
```

**Comment by MrGunflame** [comment]
Yeah, that is not correct, especially considering that `timestamp` does a 22 bit right shift.

**Comment by MrGunflame** [comment]
Fixed in bf1ebe54304f9c2eda316be66d3d2ce9be365bd2, released in `1.0.2`.
